python/lsst/daf/butler/registry/databases/postgresql.py

```diff
@@ -194,6 +197,22 @@ def _convertExclusionConstraintSpec(
             name=self.shrinkDatabaseEntityName("_".join(names)),
         )
 
+    def _make_temporary_table(
+        self,
+        connection: sqlalchemy.engine.Connection,
+        spec: ddl.TableSpec,
+        name: Optional[str] = None,
+        **kwargs: Any,
+    ) -> sqlalchemy.schema.Table:
+        # Docstring inherited
+        # Adding ON COMMIT DROP here is really quite defensive: we already
+        # manually drop the table at the end of the temporary_table context
+        # manager, and that will usually happen first.  But this will guarantee
```

Comment:
Hmm, I'm not sure where do we drop temporary tables, I don't think that new `temporary_table` does that?

Reply:
Actually, nevermind, for now - if I attempt the `drop` in the `finally`, it's apparently easy to trigger an error in which we end up trying a DROP TABLE in a read-only transaction. Will investigate tomorrow.

Reply:
Fixed, in a new fixup commit that reworks the `transaction` method inheritance a bit; I was previously running the `SET TRANSACTION READ ONLY` bit within do-nothing `transaction` context manager calls, not just the outermost "real" transaction.